llvmlite/tests/test_binding.py

```diff
@@ -1935,6 +1981,60 @@ def test_get_section_content(self):
                 self.assertEqual(s.data().hex(), issue_632_text)
 
 
+class TestArchiveFile(BaseTest):
+    mod_archive_asm = """
+        ;ModuleID = <string>
+        target triple = "{triple}"
+
+        declare i32 @__multiply_accumulate(i32 %0, i32 %1, i32 %2)
+        declare i32 @__multiply_subtract(i32 %0, i32 %1, i32 %2)
+    """
+
+    @unittest.skipUnless(sys.platform.startswith('linux'),
+                         "Linux-specific test")
+    def test_add_archive(self):
+        if not external_compiler_works():
+            self.skipTest()
+
+        tmpdir = TemporaryDirectory()
+        try:
+            target_machine = self.target_machine(jit=False)
+
+            jit = llvm.create_mcjit_compiler(self.module(self.mod_archive_asm),
+                                             target_machine)
+
+            # Create compiler with default options
+            c = new_compiler()
+            customize_compiler(c)
+            workdir = tmpdir.name
+            srcdir = os.path.dirname(llvmlite.tests.__file__) + "/"
```

Comment:
Please use `os.path.join` to make this portable and same for the path manipulation below and don't add the `.c` files to the repository. Store the code as inline strings and write them to a temporary directory during the test.